src/System.Threading.ThreadPool/tests/ThreadPoolTests.cs

```diff
@@ -349,6 +349,8 @@ public static void QueueRegisterNegativeTest()
                 ThreadPool.RegisterWaitForSingleObject(waitHandle, callback, null, -2, true));
             Assert.Throws<ArgumentOutOfRangeException>(() =>
                 ThreadPool.RegisterWaitForSingleObject(waitHandle, callback, null, (long)-2, true));
+            Assert.Throws<ArgumentOutOfRangeException>(() =>
```

Comment:
Good point. I will add it.

Reply:
For some of the `ArgumentNullException`s CoreRT and CoreCLR return different values. CoreRT correctly returns the actual parameter names (eg. "callBack"). CoreCLR mimics the legacy value from .NET Framework even though it's technically wrong ("WaitOrTimerCallback").

I'll add it to places where it's not a problem, but I would prefer the rest to be addressed separately because it will either need exceptions for certain frameworks or modifying the behavior of CoreRT or CoreCLR.

Reply:
We should ideally fix CoreCLR's name - it's OK to correct it.

Reply:
I'll submit the CoreCLR fix and amend this PR.